File: cpp/0455分发饼干.cpp
#include<vector>
#include<algorithm>
using namespace std;

int findContentChildren(vector<int> &g, vector<int> &s)
{
    sort(g.begin(), g.end());
    sort(s.begin(), s.end());
    int count = 0;
    for (int i = g.size() - 1, j = s.size(); i >=0 &&j >=0 ; --i)
    {
        if(g[i]<=s[j])
        {
            ++count;
            --j;
        }
    }

    return count;
}

File: python/0455分发饼干.py

g=[1,2,3]
s=[1,1]

def findContentChildren(g: list, s: list) -> int:
    g=sorted(g,reverse=True)
    s=sorted(s,reverse=True)

    count=0
    i=0
    j=0
    while(i<len(g) and j<len(s)):
        if(g[i]<=s[j]):
            count+=1
            j+=1
        i+=1
    
    return count